### PR TITLE
fix: stop truncating log timestamps to avoid missing logs [WEB-1791]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Context: https://hpe-aiatscale.slack.com/archives/C02PV33GSN5/p1706886505065759

TLDR: When web fetches logs from the API we we converting the timestamp from the database to and from a JavaScript Date object. Dates only support millisecond precision, so all significant digits in the microsecond range were truncated. Due to the way we request logs, this led to ~500 microseconds worth of entries not being reported on the web LogViewer. By allowing the timestamp to be passed without a conversion step, this problem should be avoided.

## Test Plan

- Go to any trial logs page
- Download the full logs though the Download Logs modal
- Open the Network tab of your browser's inspector
- Scroll up in the LogViewer until you see a GET request to `/api/v1/trials/{trial_id}/logs?limit=100&...`
- Copy the response to that request into a text file
- Continue to scroll up in LogViewer until you see another similar GET request
- Copy the response into a separate text file
- The last entry from the first API call should be the same as the first entry from the second API call
- Cross reference with full downloaded logs that there aren't any logs missing between the two requests

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ashtonG** [comment]
Approach seems solid and i think this will fix it, but I'm holding off on approval until we have a test plan

**Comment by EmilyBonar** [comment]
@ashtonG Added a test plan

**Review by ashtonG — APPROVED** [review]
